### ISSUE TITLE
Gut 4.0.0 crashes 1 in a 10 times godot 2.1.2 when yielding in a test via yield(yield_for(time), YIELD)

### ISSUE THREAD

**Issue by PKrawczynski** [bitwes/Gut#18]
Edit:
I was confused by some very rare memory crash. It's unrelated to gut itself.

**Comment by bitwes** [comment]
I don't see anything obviously wrong with the test, and nothing that I know of should cause this bad of a crash.  I haven't tried Gut on 2.1.2 yet.  If you can provide complete sample code that causes the issue that would help.  Another thing that you can try is to run the sample code from https://github.com/bitwes/GutTests/ with the 2.1.2 engine.  There is a test script in there that runs through all the [known issues with yielding](https://github.com/bitwes/GutTests/blob/master/test/unit/test_gut_yielding.gd).  If those tests can reproduce the issue then that will give me something to go on.  I need to look into upgrading my version of the engine but don't have time to do immediately (hopefully in the next week).

I would really like to see this in an easily reproducible form in 2.1.2 so I can try on the 2.1 version I'm currently using.

**Comment by PKrawczynski** [comment]
I've ran test suit, can't reproduce it that way. I suspect either I've messed up (which I will test more tommorow), or it happens when test yields immediately after instancing large amount of scenes (which can be root of problem as godot game crashes because of memory allocation if I understand that error correctly).

**Comment by bitwes** [comment]
Is it always 0xffffffff?  If so, that makes me think that something isn't initialized yet.  Maybe something hasn't been allocated or something like that.  That might explain why you see this occasionally and would also indicate that it's probably an issue with the engine and not Gut.  The fact that you can get it without yielding to Gut makes me think that as well.  When you yielding to a signal, did you use some object other than Gut...like a Timer or something?

**Comment by bitwes** [comment]
Have you tried putting some arbitrary line of code before the yield?  That might just be enough to make Godot wait until everything is done before moving on.  Try a `gut.p` or a `pending` or just a `print`.

**Comment by PKrawczynski** [comment]
Awesome I've spent some additional time testing and it does look like error is on my side. It's very obscure and I don't yet understand it completely but i've made some stress test that doesn't involve gut and I managed to crash game :)

I'm really sorry for taking your time on this. I'd like to say however I'm in deep love with unit/integration tests right now. If it wasn't for Gut I'd probably lose a lot of hours later deeper into project scratching my head why and where it crashes.

I can't express enough how grateful I am for your hard work on this framework.
